### PR TITLE
Update HAProxy to v2.6.2

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Update HAProxy docs to v2.6 (LTS)

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -10,20 +10,24 @@ This document describes best practices for configuration and usage of [HAProxy](
 
 ![HAProxy Best Practices in TiDB](/media/haproxy.jpg)
 
+> **Note:**
+>
+> The minimum version of HAProxy that works with all versions of TiDB is v1.5. Between v1.5 and v2.1 you need to set the `post-41` option in the `mysql-check`. It is recommended to use version v2.2 or newer.
```
[the remark this answers is not in the document]
Comment (dveeden) [reply best-practices/haproxy-best-practices.md:15]: https://github.com/pingcap/docs/pull/10983

**Review by bb7133 — APPROVED** [review]
LGTM

**Comment by bb7133** [comment]
PTAL @TomShawn @qiancai

**Review by shichun-0415 — APPROVED** [review]

**Comment by qiancai** [comment]
@dveeden Would you please let us know which versions this PR should also cherry-pick? Thanks.

**Review by qiancai — APPROVED** [review]

**Comment by dveeden** [comment]
> @dveeden Would you please let us know which versions this PR should also cherry-pick? Thanks.

I've update the description to set this to merge into v6.1 and v6.3.

**Pull request merged**